app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -177,7 +178,10 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
 
             // Process the metadata of each post
             if (post?.metadata && Object.keys(post?.metadata).length > 0) {
-                const data = post.metadata;
+                let data = post.metadata;
+
+                // parsing into json since notifications are sending metadata as a string
+                data = safeParseJSON(post.metadata) as PostMetadata;
```

Comment:
Do we need similar changes on the native side? Or the native side is properly handled?

Reply:
@larkox no, seems like the metadata being sent as a string is well known from the iOS native side.

```
struct MetadataSetters {
    let metadata: String
    let reactionSetters: [[Setter]]
    let fileSetters: [[Setter]]
    let emojiSetters: [[Setter]]
```

and similar parsing of string to json is already happening in iOS native.  

```
let json = try? JSONSerialization.jsonObject(with: post.metadata.data(using: .utf8)!, options: [])
```

Thus the reason, if the app is terminated, and since native iOS is doing all the parsing, that's why we're not seeing the same problem as when the app is backgrounded, where javascript is doing the work.

Reply:
Great. Just to be on the safe side, have you checked Android?

Reply:
2/5.  I think Android native CustomPushNotification.kt gets the notification but if I'm tracing it down correctly, it's sending the data to JS, so should be handled by JS.